includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -420,7 +420,11 @@ public function process_response( $response, $order ) {
 
 		$order_id = WC_Stripe_Helper::is_wc_lt( '3.0' ) ? $order->id : $order->get_id();
 
-		$captured = ( isset( $response->captured ) && $response->captured ) ? 'yes' : 'no';
+		if ( 'payment_intent' === $response->object ) {
+			$captured = ( isset( $response->status ) && 'success' === $response->status ) ? 'yes' : 'no';
```

Comment:
The `PaymentIntent` object has a `charges[]` property, this code can be rewritten like this:
```php
if ( 'payment_intent' === $response->object ) {
    $response = $response->charges[0];
}
```

Of course it's better to give `$response` a more meaningful name (like `$charge`), and double-check that there are no cases where more than 1 charge is returned on the `PaymentIntent`.

This will make sure the rest of the `process_response` function keeps working, since it uses fields like `$response->balance_transaction` that aren't present in the PaymentIntent object directly.

Reply:
Good point. I undid those changes to the abstract gateway class and added them to `WC_Gateway_Stripe`, as the _abstract_ `WC_Stripe_Payment_Gateway` should not be aware of intents.